neqo-crypto/src/lib.rs

```diff
@@ -175,23 +136,48 @@ pub fn init_db<P: Into<PathBuf>>(dir: P) -> Res<()> {
             )
         })?;
 
-        secstatus_to_res(unsafe { nss::NSS_SetDomesticPolicy() })?;
-        secstatus_to_res(unsafe {
-            p11::NSS_SetAlgorithmPolicy(
-                p11::SECOidTag::SEC_OID_XYBER768D00,
-                p11::NSS_USE_ALG_IN_SSL_KX,
-                0,
-            )
-        })?;
         secstatus_to_res(unsafe {
             ssl::SSL_ConfigServerSessionIDCache(1024, 0, 0, dircstr.as_ptr())
         })?;
+        NssLoaded::Db
+    } else {
+        secstatus_to_res(unsafe { nss::NSS_NoDB_Init(null()) })?;
+        NssLoaded::NoDb
+    };
+
+    secstatus_to_res(unsafe { nss::NSS_SetDomesticPolicy() })?;
+    secstatus_to_res(unsafe {
+        p11::NSS_SetAlgorithmPolicy(
+            p11::SECOidTag::SEC_OID_XYBER768D00,
+            p11::NSS_USE_ALG_IN_SSL_KX,
+            0,
+        )
+    })?;
+
+    #[cfg(debug_assertions)]
+    enable_ssl_trace()?;
```

Comment:
I assume it is fine that we do this for `init` and `init_db` now, whereas previously we only did it for `init_db`.

Reply:
Yeah, it didn't seem like this was a useful difference, so I nuked it.  Obviously, the session cache thing doesn't work that way, so that is only available if we have a database.